src/components/plots/abstractXYPlot.ts

```diff
@@ -115,65 +137,55 @@ module Plottable {
       return attrToProjector;
     }
 
-    public _computeLayout(offeredXOrigin?: number, offeredYOffset?: number, availableWidth?: number, availableHeight?: number) {
-      super._computeLayout(offeredXOrigin, offeredYOffset, availableWidth, availableHeight);
-      this._xScale.range([0, this.width()]);
-      if (this._yScale instanceof Scales.Category) {
-        this._yScale.range([0, this.height()]);
-      } else {
-        this._yScale.range([this.height(), 0]);
-      }
+    protected normalizeDatasets<A, B>(fromX: boolean): {a: A; b: B}[] {
+      var aAccessor: (d: any, i: number, u: any, m: Plots.PlotMetadata) => A = this.projections[fromX ? "x" : "y"].accessor;
+      var bAccessor: (d: any, i: number, u: any, m: Plots.PlotMetadata) => B = this.projections[fromX ? "y" : "x"].accessor;
+      return Utils.Methods.flatten(this.datasetKeysInOrder.map((key: string) => {
+        var dataset = this.datasetKeys.get(key).dataset;
+        var plotMetadata = this.datasetKeys.get(key).plotMetadata;
+        return dataset.data().map((d, i) => {
+          return { a: aAccessor(d, i, dataset.metadata(), plotMetadata), b: bAccessor(d, i, dataset.metadata(), plotMetadata) };
+        });
+      }));
     }
 
-    protected _updateXDomainer() {
+    protected projectorsReady() {
+      return this.projections["x"] && this.projections["y"];
+    }
+
+    protected updateXDomainer() {
       if (this._xScale instanceof QuantitativeScale) {
         var scale = <QuantitativeScale<any>> this._xScale;
-        if (!scale._userSetDomainer) {
+        if (!scale.setByUser) {
```

Comment:
Rename this back to userSetDomainer.  We will have a relook at this later.  Let's for now keep this scope as minimal as possible